app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseManagementTest.kt

```diff
@@ -0,0 +1,285 @@
+        val name = getRandomString()
+        val ds = scratchManagement.putDataset(name, setOf())
+        assertThat(scratchManagement.getDatasetById(ds.id)!!.name, equalTo(name))
+    }
+
+    fun test_getDatasetById() = runBlocking {
+        assertThat(demoManagement.getDatasetById("PzuR0B48GpYN5ERxM3DW"), not(equalTo(null)))
+    }
+
+    fun test_getDatasetByName() = runBlocking {
+        assertThat(demoManagement.getDatasetByName("Fruit").size, equalTo(1))
+    }
+
+    fun test_deleteDataset() = runBlocking {
+        val ds = scratchManagement.putDataset(getRandomString(), setOf())
+        requireNotNull(scratchManagement.getDatasetById(ds.id))
+        scratchManagement.deleteDataset(ds.id)
+        assertThat(scratchManagement.getDatasetById(ds.id), equalTo(null))
+    }
+
+    fun test_putRepresentativePicture_categoryNotPresent() = runBlocking {
+        runCatching {
+            scratchManagement.getRepresentativePicture(fakeCategory)
+        }.fold({
+            fail("unexpected successful completion")
+        }, {
+            assertThat(it, instanceOf(IllegalArgumentException::class.java))
+        })
+    }
+
+    fun test_putRepresentativePicture() = runBlocking {
+        val randomCategoryName = getRandomString()
+        val ctx = ApplicationProvider.getApplicationContext<Context>()
+        val cat = scratchManagement.putCategory(randomCategoryName)
+        val tmp = File.createTempFile("droid", ".png")
+        try {
+            ctx.resources.openRawResource(R.drawable.fork).use { img ->
+                tmp.outputStream().use {
+                    img.copyTo(it)
+                }
+            }
+            val uri = Uri.fromFile(tmp)
+            scratchManagement.putRepresentativePicture(uri, cat)
+        } finally {
+            tmp.delete()
+        }
+
+        assertThat(scratchManagement.getRepresentativePicture(cat.id), not(equalTo(null)))
+    }
+
+    fun test_getDatasets() = runBlocking {
+        assertThat(demoManagement.getDatasets().size, equalTo(1))
+        assertThat(demoManagement.getDatasets().first().name, equalTo("Fruit"))
+    }
+
+    fun test_getDatasetNames() = runBlocking {
+        assertThat(demoManagement.getDatasetNames().size, equalTo(1))
+        assertThat(demoManagement.getDatasetNames().first(), equalTo("Fruit"))
+    }
+
+    fun test_getDatasetIds() = runBlocking {
+        assertThat(demoManagement.getDatasetIds().size, equalTo(1))
+        assertThat(demoManagement.getDatasetIds().first(), equalTo("PzuR0B48GpYN5ERxM3DW"))
+    }
+
+    fun test_removeCategoryFromDataset_datasetNotPresent() = runBlocking {
+        val cat1 = scratchManagement.putCategory(getRandomString())
+        val cat2 = scratchManagement.putCategory(getRandomString())
+        val fakeDs = FirestoreDataset("path", getRandomString(), getRandomString(), setOf(cat1, cat2))
+        runCatching {
+            scratchManagement.removeCategoryFromDataset(fakeDs, cat2)
```

Comment:
It's probably because I'm not very familiar with the API, but why should removing an existing category from the dataset throw an exception?

Reply:
It should not indeed for the same reasons the other remove method do not. Thank you